src/olympia/files/tests/test_models.py

```diff
@@ -420,11 +421,39 @@ def test_parse_permissions(self):
             u'https://google.com/']
 
     def test_parse_apps(self):
-        exp = (amo.FIREFOX,
-               amo.FIREFOX.id,
-               AppVersion.objects.get(version='3.0'),
-               AppVersion.objects.get(version='3.6.*'))
-        assert self.parse()['apps'] == [exp]
+        expected = [Extractor.App(
+            amo.FIREFOX, amo.FIREFOX.id,
+            AppVersion.objects.get(version='3.0'),
+            AppVersion.objects.get(version='3.6.*'))]
+        assert self.parse()['apps'] == expected
+
+    def test_parse_max_star(self):
+        AppVersion.objects.create(application=amo.FIREFOX.id, version='56.*')
+        AppVersion.objects.create(application=amo.FIREFOX.id, version='*')
+        AppVersion.objects.create(application=amo.FIREFOX.id, version='38.0a1')
+        AppVersion.objects.create(application=amo.ANDROID.id, version='*')
+        AppVersion.objects.create(application=amo.ANDROID.id, version='56.*')
+        AppVersion.objects.create(application=amo.ANDROID.id, version='38.0a1')
+
+        # The install.rdf in jetpack_star.xpi is using '*' as the max version,
+        # but it should be rewritten to '56.*'.
+        expected = [
+            Extractor.App(
+                amo.FIREFOX, amo.FIREFOX.id,
+                AppVersion.objects.get(
+                    application=amo.FIREFOX.id, version='38.0a1'),
+                AppVersion.objects.get(
+                    application=amo.FIREFOX.id, version='*')),
+
+            Extractor.App(
+                amo.ANDROID, amo.ANDROID.id,
+                AppVersion.objects.get(
+                    application=amo.ANDROID.id, version='38.0a1'),
+                AppVersion.objects.get(
+                    application=amo.ANDROID.id, version='*'))]
```

Comment:
I might be blind but according to the comment this should be `version='56.*'` right?